pydantic/validate_call.py

```diff
@@ -43,6 +43,14 @@ def validate_call(
     """
 
     def validate(function: AnyCallableT) -> AnyCallableT:
+        if isinstance(function, classmethod):
+            raise TypeError(
+                'The `@classmethod` decorator should be applied after `@validate_call` (put `@classmethod` on top)'
+            )
+        elif isinstance(function, staticmethod):
+            raise TypeError(
+                'The `@staticmethod` decorator should be applied after `@validate_call` (put `@staticmethod` on top)'
+            )
```

Comment:
```suggestion
        if isinstance(function, (classmethod, staticmethod)):
            name = type(function).__name__
            raise TypeError(
                f'The `@classmethod` decorator should be applied after `@{name}` (put `@{name}` on top)'
            )
```
Could do this instead, not sure which is preferable.